src/array/methods.jl

```diff
@@ -104,17 +104,48 @@ function Base.mapslices(f, A::AbstractDimArray; dims=1, kw...)
     rebuild(A, data)
 end
 
-# This is copied from base as we can't efficiently wrap this function
-# through the kw with a rebuild in the generator. Doing it this way
-# also makes it faster to use a dim than an integer.
-function Base.eachslice(A::AbstractDimArray; dims=1, kw...)
```

Comment:
I removed the default `dims=1` because this is inconsistent with base. But perhaps this is a breaking change?

Reply:
Its more that its a bug to be different to Base. So I think that's fine.